### PR TITLE
Fix entry count massa module cache

### PR DESCRIPTION
**Pull request by Leo-Besancon**
* [ ] document all added functions
* [ ] try in sandbox /simulation/labnet
  * [ ] if part of node-launch, checked using the `resync_check` flag
* [ ] unit tests on the added/changed features
  * [ ] make tests compile
  * [ ] make tests pass 
* [ ] add logs allowing easy debugging in case the changes caused problems
* [ ] if the API has changed, update the API specification

Note: as we currently clear the cache in main when the node starts, this does not impact the current code.
It's only fixed to avoid potential future errors if this is changed.

As a result, this does not need to be versioned.

### CODE REVIEW

File: massa-module-cache/src/hd_cache.rs [hunk 1]
```diff
@@ -58,9 +58,22 @@ impl HDCache {
...
     /// * max_entry_count: maximum number of entries we want to keep in the db
     /// * amount_to_remove: how many entries are removed when `entry_count` reaches `max_entry_count`
+    ///
+    /// Note: entry_count refers to the number of modules stored in the db, which is half the number of (key, value) pairs in the db
     pub fn new(path: PathBuf, max_entry_count: usize, snip_amount: usize) -> Self {
-        let db = DB::open_default(path).expect(OPEN_ERROR);
-        let entry_count = db.iterator(IteratorMode::Start).count();
+        let mut db = DB::open_default(path.clone()).expect(OPEN_ERROR);
+        let mut nb_db_entries = db.iterator(IteratorMode::Start).count();
+        if nb_db_entries % 2 != 0 {
```
Comment (modship) [inline massa-module-cache/src/hd_cache.rs:66]: maybe do with it with checked_div ?

Comment (damip) [reply massa-module-cache/src/hd_cache.rs:66]: checked_div does not check for divisibility, only for divisions by 0

File: massa-module-cache/src/hd_cache.rs [hunk 2]
```diff
@@ -58,9 +58,22 @@ impl HDCache {
     /// * path: where to store the db
     /// * max_entry_count: maximum number of entries we want to keep in the db
     /// * amount_to_remove: how many entries are removed when `entry_count` reaches `max_entry_count`
+    ///
+    /// Note: entry_count refers to the number of modules stored in the db, which is half the number of (key, value) pairs in the db
     pub fn new(path: PathBuf, max_entry_count: usize, snip_amount: usize) -> Self {
-        let db = DB::open_default(path).expect(OPEN_ERROR);
-        let entry_count = db.iterator(IteratorMode::Start).count();
+        let mut db = DB::open_default(path.clone()).expect(OPEN_ERROR);
+        let mut nb_db_entries = db.iterator(IteratorMode::Start).count();
```
Comment (damip) [inline massa-module-cache/src/hd_cache.rs:65]: this count might be super heavy (?)

**Comment by Leo-Besancon** [comment]
Updated to simply clear cache in new, and setting the entry_count to 0.

**Pull request merged**